doc/decisions/0_drafts/errors_core.md

```diff
@@ -352,6 +352,37 @@ This is allowed, because it has the same expressiveness and use case as a `Key *
 
 Internal errors that are never the fault of the caller and always caused by a bug in Elektra, should be prevented with an `assert`.
 
+#### Secondary return values
+
+In the success case, only return data the caller explicitly asked for by calling the function.
+Do not use secondary information as the return value.
+If there is secondary information, the caller may care about, use an (optional) output pointer argument.
```

Comment:
I don't understand what you want to say here. What is a "secondary return"?

Reply:
Basically, that the `keySetName` API (and almost all other APIs that return `ssize_t`) is stupid.

There is no reason to return a size. The caller did not ask for a size. They wanted to change the name. That's an operation with no inherent result, so in theory it should be `void keySetName (...)`. But it can fail, so we should `int keySetName (...)`.

The bit about secondary information is meant to say: If the new size really is valuable information then it should be returned differently, since it is only secondary information and not the primary result of the function:

```c
// on success, if outNewSize != NULL it will be set to the new size of key's name
int keyAddName (Key * key, const char * suffix, size_t * outNewSize);

// if outNewSize != NULL it will be set to the new size of ks
void ksAppendKey (KeySet * ks, Key * key, size_t * outNewSize);
```